packages/create-cloudflare/src/common.ts

```diff
@@ -250,33 +253,74 @@ export const offerGit = async (ctx: PagesGeneratorContext) => {
 };
 
 export const gitCommit = async (ctx: PagesGeneratorContext) => {
-	if (!ctx.args.git) return;
+	if (!(await isGitInstalled()) || !(await isInsideGitRepo(ctx.project.path)))
+		return;
+
+	const commitMessage = await createCommitMessage(ctx);
 
 	await runCommands({
 		silent: true,
 		cwd: ctx.project.path,
-		commands: [
-			"git add .",
-			["git", "commit", "-m", "Initial commit (by Create-Cloudflare CLI)"],
-		],
+		commands: ["git add .", ["git", "commit", "-m", commitMessage]],
 		startText: "Committing new files",
-		doneText: `${brandColor("git")} ${dim(`initial commit`)}`,
+		doneText: `${brandColor("git")} ${dim(`commit`)}`,
 	});
 };
 
+const createCommitMessage = async (ctx: PagesGeneratorContext) => {
+	if (!ctx.framework) return "Initial commit (by create-cloudflare CLI)";
+
+	const header = "Initialize web application via create-cloudflare CLI";
+
+	const packageManager = detectPackageManager();
+
+	const gitVersion = await getGitVersion();
+	const insideRepo = await isInsideGitRepo(ctx.project.path);
+
+	const details = [
+		{ key: "C3", value: `create-cloudflare@${version}` },
+		{ key: "project name", value: ctx.project.name },
+		{ key: "framework", value: ctx.framework.name },
+		{ key: "framework cli", value: getFrameworkCli(ctx) },
+		{
+			key: "package manager",
+			value: `${packageManager.name}@${packageManager.version}`,
+		},
+		{
+			key: "wrangler",
+			value: `wrangler@${wranglerVersion}`,
+		},
+		{
+			key: "git",
+			value: insideRepo ? gitVersion : "N/A",
+		},
+	];
+
+	const body = `Details:\n${details
+		.map(({ key, value }) => `  ${key} = ${value}`)
+		.join("\n")}\n`;
+
+	return `${header}\n\n${body}\n`;
+};
+
 /**
- * Check whether git is available on the user's machine.
+ * Return the version of git on the user's machine, or null if git is not available.
  */
-export async function isGitInstalled() {
+async function getGitVersion() {
 	try {
-		await runCommand("git -v", { useSpinner: false, silent: true });
-
-		return true;
+		return runCommand("git -v", { useSpinner: false, silent: true });
```

Comment:
on my ubuntu (20.04.2 LTS):

![Screenshot at 2023-08-16 18-38-30](https://github.com/cloudflare/workers-sdk/assets/61631103/17dc4b23-2deb-472d-b9e4-b35f482d09e1)

so I think we should use `--version` to be on the safe side :thinking:

Reply:
yes